components/bms_worker/include/BatteryMonitoring.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file BatteryMonitoring.h
+ * @brief  Header file for Battery Monitoring Application
+ * @author Joshua Lafleur (josh.lafleur@outlook.com)
+ * @date 2023-12-27
+ */
+
+#pragma once
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+/**< System Includes */
+#include "stdint.h"
+#include "stdbool.h"
+
+/**< Firmware Includes */
+#include "HW_adc.h"
+
+/**< Driver Includes */
+#include "HW_MAX14921.h"
+
+/******************************************************************************
+ *                              D E F I N E S
+ ******************************************************************************/
+
+#define BMS_ADC_BUF_LEN 264
+
+
+/******************************************************************************
+ *                              E X T E R N S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef enum {
+    BMS_INIT = 0x00,
+    BMS_CALIBRATING,
+    BMS_WAITING,
+    BMS_SAMPLING,
+    BMS_PACK_SAMPLING,
+    BMS_DIAGNOSTIC,
+    BMS_BALANCING,
+    BMS_ERROR,
+} BMS_State_E;
+
+typedef enum
+{
+    ADC_CELL_MEASUREMENT = 0x00,
+    ADC_CHANNEL_COUNT,
+} AdcChannels_E;
+
+typedef struct {
+    bool measurement_complete;
+    uint32_t adc_buffer[BMS_ADC_BUF_LEN];
+    simpleFilter_S cell_voltages[ADC_CHANNEL_COUNT];
+    simpleFilter_S pack_voltage;
+} BMS_Data_S;
+
+typedef struct {
+    BMS_State_E state;
+    uint16_t balancing_cells;
+    uint16_t cell_faults;
+    uint16_t cell_voltages[CELL_COUNT];
+    uint16_t pack_voltage; /**< Stored in 1mV units */
```

Comment:
should use something like this for units:
```suggestion
    uint16_t pack_voltage; // [mV] pack voltage, precision of 1mV
```